compute/instances/preventing-accidental-vm-deletion/preventing_accidental_vm_deletion_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+	"time"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+)
+
+func TestPreventingAccidentalVMDeletionSnippets(t *testing.T) {
+	ctx := context.Background()
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		t.Fatalf("NewInstancesRESTClient: %v", err)
+	}
+	defer instancesClient.Close()
+
+	var seededRand *rand.Rand = rand.New(
+		rand.NewSource(time.Now().UnixNano()))
+	tc := testutil.SystemTest(t)
+	zone := "europe-central2-b"
+	instanceName := "test-instance-" + fmt.Sprint(seededRand.Int())
+
+	buf := &bytes.Buffer{}
+
+	if err := createInstance(buf, tc.ProjectID, zone, instanceName, true); err != nil {
+		t.Fatalf("createInstance got err: %v", err)
+	}
+
+	expectedResult := "Instance create"
```

Comment:
The string matching here is probably unnecessary. A nil error return is adequate to ensure the operation worked. Otherwise, consider returning a value that could be checked instead.

Reply:
Also, prefer `got` instead of `expectedResult`.

Reply:
If we wanted a different name here I think `want` would be the correct shorter name.

Reply:
:man_facepalming: yes, `want`. Woops.